tests/instron_stress_rig.py

```diff
@@ -0,0 +1,99 @@
+        self.ca.assert_that_pv_is("INSTRON_01:GOING", "NO")
+        self.ca.set_pv_value("INSTRON_01:MOVE:GO:SP", 1)
+        self.ca.assert_that_pv_is("INSTRON_01:GOING", "YES")
+        self.ca.set_pv_value("INSTRON_01:STOP:SP", 1)
+        self.ca.assert_that_pv_is("INSTRON_01:GOING", "NO")
+
+    @skipIf(IOCRegister.uses_rec_sim, "In rec sim this test fails")
+    def test_that_going_and_then_panic_stopping_reflects_the_expected_state(self):
+        self.ca.assert_that_pv_is("INSTRON_01:GOING", "NO")
+        self.ca.set_pv_value("INSTRON_01:MOVE:GO:SP", 1)
+        self.ca.assert_that_pv_is("INSTRON_01:GOING", "YES")
+        self.ca.set_pv_value("INSTRON_01:PANIC:SP", 1)
+        self.ca.assert_that_pv_is("INSTRON_01:GOING", "NO")
+
+    @skipIf(IOCRegister.uses_rec_sim, "In rec sim this test fails")
+    def test_that_an_arbitrary_command_Q22_can_be_sent(self):
+        self.ca.set_pv_value("INSTRON_01:ARBITRARY:SP", "Q22")
+        # Assert that the response to Q22 is a status code
+        self.ca.assert_that_pv_is_an_integer_between("INSTRON_01:ARBITRARY", min=0, max=65535)
+
+    @skipIf(IOCRegister.uses_rec_sim, "In rec sim this test fails")
+    def test_that_an_arbitrary_command_Q300_can_be_sent(self):
+        self.ca.set_pv_value("INSTRON_01:ARBITRARY:SP", "Q300")
+        # Assert that the response to Q300 is between 1 and 3
+        self.ca.assert_that_pv_is_an_integer_between("INSTRON_01:ARBITRARY", min=1, max=3)
+
+    @skipIf(IOCRegister.uses_rec_sim, "In rec sim this test fails")
+    def test_that_an_arbitrary_command_C1_and_Q1_act_as_expected(self):
+
+        def _set_and_check(value):
+            self.ca.set_pv_value("INSTRON_01:ARBITRARY:SP", "C1," + value)
+            self.ca.assert_that_pv_is("INSTRON_01:ARBITRARY:SP", "C1," + value)
+            self.ca.set_pv_value("INSTRON_01:ARBITRARY:SP", "Q1")
+            self.ca.assert_that_pv_is("INSTRON_01:ARBITRARY", value)
+
+        for v in ["0", "1", "0"]:
+            _set_and_check(v)
+
+    @skipIf(IOCRegister.uses_rec_sim, "In rec sim this test fails")
+    def test_that_if_the_movement_type_on_rig_is_hold_then_it_gets_stopped(self):
+
+        self.ca.set_pv_value("INSTRON_01:MOVE:SP", 1)
+        self.ca.assert_that_pv_is_one_of("INSTRON_01:MOVE", ["RAMP_RUNNING", "RAND_RUNNING"])
+
+        self.ca.set_pv_value("INSTRON_01:MOVE:SP", 2)
+        self.ca.assert_that_pv_is("INSTRON_01:MOVE", "STOPPED")
+
+    def test_that_control_channel_is_one_of_the_allowed_values(self):
+        self.ca.assert_that_pv_is_one_of("INSTRON_01:CHANNEL", ["Stress", "Strain", "Position"])
+
+    def test_that_control_channel_setpoint_is_one_of_the_allowed_values(self):
+        self.ca.assert_that_pv_is_one_of("INSTRON_01:CHANNEL:SP", ["Stress", "Strain", "Position"])
+
+    @skipIf(IOCRegister.uses_rec_sim, "In rec sim this test fails")
+    def test_that_setting_the_control_channel_works_as_expected(self):
```

Comment:
Good tests, thanks. As discussed. Please reconsider using test_GIVEN_context_WHEN_action_THEN_result. It seems especially relevant to emulators since devices may have very specific contexts. As the stress rig gets more complicated, the actual situation under test will get much less clear without a very obvious context.